internal/plugin/util/schema_converter.go

```diff
@@ -0,0 +1,120 @@
+package util
+
+import (
+	"fmt"
+
+	datasource_schema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+// ResourceSchemaToDataSourceSchema converts a resource schema to a data source schema.
+// All attributes are made computed by default, except for the specified required fields.
+// See https://github.com/hashicorp/terraform-plugin-framework/issues/568.
+// Adapted from https://github.com/Doridian/terraform-provider-hexonet/blob/42a3ee1ab12544cde4d31b4897ceacf75d46cb3a/hexonet/utils/schema_helpers.go#L10.
+func ResourceSchemaToDataSourceSchema(resourceSchema map[string]resource_schema.Attribute, requiredFields ...string) map[string]datasource_schema.Attribute {
```

Comment:
I don't know if it covers all possible cases. I can [see](https://github.com/hashicorp/terraform-plugin-framework/blob/main/resource/schema/attribute.go#L10) the list is much bugger for types that implements `Attribute` interface.

Reply:
It probably doesn't handle all cases. We can add more later if needed.

Reply:
Another option posted to the linked issue: https://github.com/hashicorp/terraform-plugin-framework/issues/673#issuecomment-2727116463